imgparse/imgparse.py

```diff
@@ -244,17 +244,33 @@ def _get_terrain_elevation(image_path, exif_data, xmp_data, api_key):
     """
     home_lat, home_lon = get_home_point(image_path, exif_data, xmp_data)
     image_lat, image_lon = get_lat_lon(image_path, exif_data)
+    home_elevation = _get_home_point_elevation(home_lat, home_lon, api_key)
+    image_elevation = _get_terrain_elevation(image_lat, image_lon, api_key)
+    return home_elevation - image_elevation
+
+
+def _get_terrain_elevation(lat, lon, api_key):
+    """Call out to the google elevation api to get the terrain elevation at a given lat/lon."""
     params = {
-        "locations": f"{home_lat} {home_lon} | {image_lat} {image_lon}",
+        "locations": f"{lat} {lon}",
         "key": api_key,
     }
     response = requests.request("GET", TERRAIN_URL, params=params).json()
     if response["status"] != "OK":
         logger.warning("Couldn't access google terrain api")
         raise ParsingError("Couldn't access google terrain api")
-    home_elevation = response["results"][0]["elevation"]
-    image_elevation = response["results"][1]["elevation"]
-    return home_elevation - image_elevation
+    return response["results"][0]["elevation"]
+
+
+@memoize
+def _get_home_point_elevation(lat, lon, api_key):
+    """
+    Memoized wrapper around `get_terrain_elevation()`.
+
+    Ensure we aren't making multiple calls to google to get the same home point elevation for images from the
+    same flight.
+    """
+    return _get_terrain_elevation(lat, lon, api_key)
```

Comment:
This looks _super_ nice. My first thought when I read this though was that you could just `@memoize` the `_get_terrain_elevation` function itself, but I'm assuming you don't do this because you'll probably just have a _ton_ of cache misses and the memoization will be wasted.

Reply:
Yeah the main reason is the memoize decorator can only [store a single item in its cache](https://github.com/SenteraLLC/py-image-metadata-parser/blob/51db9675c6ead5e0efc9701721c33c48731d219b/imgparse/decorators.py#L104-L107), so we would clear the cached home point with every subsequent query for the image's home point